src/mk/def_rules.mk

```diff
@@ -933,7 +933,9 @@ define CMDSRC_TO_CMD_RECIPE =
 	$(eval d = $($@_d))
 	@$(call echo_cmd,"=== Creating command [$(notdir $<)] in $(OBJLIB)")
 	$(eval crtcmd := CRTCMD CMD($(OBJLIB)/$(basename $(@F))) srcstmf('$<') $(CRTCMDFLAGS))
-	$(eval logFile := $(LOGPATH)/$(notdir $(basename $<)).splf)
+	$(eval directory := $(subst /,_,$(patsubst $(SRCPATH)/%,%,$(dir $<)))) 
+	$(eval extension := $(patsubst .%,%,$(suffix $<)))
+	$(eval logFile := $(LOGPATH)/$(directory)$(notdir $(basename $<))_$(extension).splf)
```

Comment:
could these three lines be move to PRESETUP so they do not have to be repeated everywhere

Reply:
Some changes needed to be made but yes